atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/KvsSweepQueueTest.java

```diff
@@ -0,0 +1,243 @@
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_NOTHING, TS), TS);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_NOTHING, TS2), TS2);
+        verify(kvs, times(0)).put(eq(TargetedSweepTableFactory.of().getSweepableCellsTable(null).getTableRef()),
+                anyMap(), anyLong());
+        verify(kvs, times(0)).put(eq(TargetedSweepTableFactory.of().getSweepableTimestampsTable(null).getTableRef()),
+                anyMap(), anyLong());
+        verify(kvs, times(0)).put(eq(TargetedSweepTableFactory.of().getSweepShardProgressTable(null).getTableRef()),
+                anyMap(), anyLong());
+    }
+
+    @Test
+    public void conservativeSweepAddsSentinelAndLeavesSingleValue() {
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONSERVATIVE, TS), TS);
+        assertNothing(TABLE_CONSERVATIVE, TS);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSentinel(TABLE_CONSERVATIVE, TS);
+        assertValue(TABLE_CONSERVATIVE, TS + 1, TS);
+    }
+
+    @Test
+    public void thoroughSweepDoesNotAddSentinelAndLeavesSingleValue() {
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_THOROUGH, TS), TS);
+        assertNothing(TABLE_THOROUGH, TS);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.thorough(THOR_SHARD));
+        assertNothing(TABLE_THOROUGH, TS);
+        assertValue(TABLE_THOROUGH, TS + 1, TS);
+    }
+
+    @Test
+    public void conservativeSweepDeletesLowerValue() {
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONSERVATIVE, TS), TS);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_CONSERVATIVE, TS2), TS2);
+        assertValue(TABLE_CONSERVATIVE, TS + 1, TS);
+        assertValue(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSentinel(TABLE_CONSERVATIVE, TS + 1);
+        assertValue(TABLE_CONSERVATIVE, TS2 + 1, TS2);
+    }
+
+    @Test
+    public void thoroughSweepDeletesLowerValue() {
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_THOROUGH, TS), TS);
+        sweepQueue.enqueue(writeToDefaultCell(TABLE_THOROUGH, TS2), TS2);
+        assertValue(TABLE_THOROUGH, TS + 1, TS);
+        assertValue(TABLE_THOROUGH, TS2 + 1, TS2);
+
+        sweepQueue.sweepNextBatch(ShardAndStrategy.thorough(THOR_SHARD));
+        assertNothing(TABLE_THOROUGH, TS + 1);
+        assertValue(TABLE_THOROUGH, TS2 + 1, TS2);
+    }
+
+    @Test
+    public void sweepDeletesAllButLatestWithSingleDeleteAllTimestamps() {
+        long numWrites = 1000L;
+        for (long i = 1; i <= numWrites; i++) {
+            sweepQueue.enqueue(writeToDefaultCell(TABLE_CONSERVATIVE, i), i);
+        }
+        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        assertSentinel(TABLE_CONSERVATIVE, numWrites);
+        assertValue(TABLE_CONSERVATIVE, numWrites + 1, numWrites);
+        verify(kvs).deleteAllTimestamps(eq(TABLE_CONSERVATIVE), anyMap());
```

Comment:
Do we want to add `times(1)` (or maybe there's a `once()`) here to make it explicitly?